daqLogger/src/main.cpp

```diff
@@ -0,0 +1,191 @@
+CANRXMessage<2> rx_message_pos{can_bus, kGPS_CAN, lon_signal, lat_signal};
+
+const int CSpin = 5;
+
+File sensorData;
+String fileName;
+String dataString ="";
+
+VirtualTimerGroup timer_group;
+
+void saveData(){
+  // reopen file
+  sensorData = SD.open(fileName, FILE_WRITE);
+  if (sensorData){
+    // print line into csv
+    sensorData.println(dataString);
+    sensorData.close();
+    Serial.println(dataString);
+  } else {
+    Serial.println("Error saving values to file !");
+  }
+}
+
+void sensor10ms(){
+  DateTime now = rtc.now();
+
+  dataString = dataString + "\n" + now.timestamp() + ", , " + float(accel_x) + ", " + float(accel_y) + ", " + float(accel_z) + ", , , , , ";
+  saveData();
+}
+
+void sensor100ms(){
+  DateTime now = rtc.now();
+
+  dataString = dataString + "\n" + now.timestamp() + float(wheel_speed_signal) + ", , " + float(accel_x) + ", " + float(accel_y) + ", " + float(accel_z) + ", " + float(gyro_x) + ", " + float(gyro_y) + ", " + float(gyro_z) + ", " + float(lon_signal) + ", " + float(lat_signal);
+  saveData();
+}
+
+void sensor1000ms(){
+  DateTime now = rtc.now();
+
+  dataString = dataString + "\n" + now.timestamp() + float(wheel_speed_signal) + ", " + float(brake_temp_signal) + ", " + float(accel_x) + ", " + float(accel_y) + ", " + float(accel_z) + ", " + float(gyro_x) + ", " + float(gyro_y) + ", " + float(gyro_z) + ", " + float(lon_signal) + ", " + float(lat_signal);
+  saveData();
+}
+
+void setup(void){
+  Serial.begin(9600);
+
+  Serial.print("Initializing CAN...");
+  can_bus.Initialize(ICAN::BaudRate::kBaud1M);
+  Serial.print("CAN Initialized");
+
+  Serial.print("Initializing timers...");
+  timer_group.AddTimer(10U, sensor10ms);
+  timer_group.AddTimer(100U, sensor100ms);
+  timer_group.AddTimer(1000U, sensor1000ms);
+  Serial.print("Timers Initialized");
+
+  Serial.print("Initializing RTC...");
+
+  #ifndef ESP8266
+    while (!Serial); // wait for serial port to connect. Needed for native USB
+  #endif
+
+  if (! rtc.begin()) {
+    Serial.println("Couldn't find RTC");
+    Serial.flush();
+    while (1) delay(10);
+  }
+
+  if (! rtc.initialized() || rtc.lostPower()) {
+    Serial.println("RTC is NOT initialized, let's set the time!");
+    rtc.adjust(DateTime(F(__DATE__), F(__TIME__)));
+  }
+
+  rtc.start();
+  DateTime now = rtc.now();
+
+  float drift = 43; // seconds plus or minus over oservation period - set to 0 to cancel previous calibration.
+  float period_sec = (7 * 86400);  // total obsevation period in seconds (86400 = seconds in 1 day:  7 days = (7 * 86400) seconds )
+  float deviation_ppm = (drift / period_sec * 1000000); //  deviation in parts per million (μs)
+  float drift_unit = 4.34; // use with offset mode PCF8523_TwoHours
+  // float drift_unit = 4.069; //For corrections every min the drift_unit is 4.069 ppm (use with offset mode PCF8523_OneMinute)
+  int offset = round(deviation_ppm / drift_unit);
```

Comment:
could you look into if we could calibrate this so that it's the time starting from like the start of 2023 rather than the standard start time?

Reply:
it would make the value smaller so we can use less bytes